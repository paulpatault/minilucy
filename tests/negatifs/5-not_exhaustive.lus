type t = A | B | C

node main (x : t) returns (o: bool);
let
  o = merge x
        (A't -> true  when A(x))
        (B't -> true  when B(x));
tel
