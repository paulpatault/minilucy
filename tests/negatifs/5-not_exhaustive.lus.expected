
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True
type t =
  | A
  | B
  | C

node main(x : adt(t)) returns (o : bool)
var ;
let
  (o) = merge x 
          (t(A) -> true when A(x))
          (t(B) -> true when B(x));
tel

